### PR TITLE
Issue #325 and #326 - Add support for base64 configuration

### PR DESCRIPTION
**Pull request by katholen**
This is the updates required for supporting base64 encoding of service/operation names.

### CODE REVIEW

File: src/main/scala/com/expedia/www/haystack/commons/entities/MetricPoint.scala [hunk 1]
```diff
@@ -28,17 +30,23 @@ import com.expedia.www.haystack.commons.entities.MetricType.MetricType
...
-        tag + s"${tuple._1}.${tuple._2}."
-      })
-    }
+    getMetricPointKey(enablePeriodReplacement, false)
+  }
+
+  def getMetricPointKey(enablePeriodReplacement: Boolean, enableBase64Encoding: Boolean): String = {
+    val metricTags = tags.foldLeft("")((tag, tuple) => {
+      var tuple2 = tuple._2
+      if ("serviceName".equalsIgnoreCase(tuple._1) || "operationName".equalsIgnoreCase(tuple._1)) {
```
Comment (ayansen) [inline src/main/scala/com/expedia/www/haystack/commons/entities/MetricPoint.scala]: can't we base64 encode all tag values ? we're decoding all the tag values in the createTagsFromMetricKey function

File: src/main/scala/com/expedia/www/haystack/commons/entities/MetricPoint.scala [hunk 2]
```diff
@@ -28,17 +30,23 @@ import com.expedia.www.haystack.commons.entities.MetricType.MetricType
   * @param epochTimeInSeconds : epochTime in seconds for when the event is generated
   */
 case class MetricPoint(metric: String, `type`: MetricType, tags: Map[String, String], value: Float, epochTimeInSeconds: Long) {
-
   def getMetricPointKey(enablePeriodReplacement: Boolean): String = {
```
Comment (ayansen) [inline src/main/scala/com/expedia/www/haystack/commons/entities/MetricPoint.scala]: we shouldn't be needing this overloaded function anymore

File: src/main/scala/com/expedia/www/haystack/commons/entities/MetricPoint.scala [hunk 3]
```diff
@@ -28,17 +30,23 @@ import com.expedia.www.haystack.commons.entities.MetricType.MetricType
...
-      })
-    }
+    getMetricPointKey(enablePeriodReplacement, false)
+  }
+
+  def getMetricPointKey(enablePeriodReplacement: Boolean, enableBase64Encoding: Boolean): String = {
+    val metricTags = tags.foldLeft("")((tag, tuple) => {
+      var tuple2 = tuple._2
+      if ("serviceName".equalsIgnoreCase(tuple._1) || "operationName".equalsIgnoreCase(tuple._1)) {
+        if (enablePeriodReplacement) {
```
Comment (ayansen) [inline src/main/scala/com/expedia/www/haystack/commons/entities/MetricPoint.scala]: Should we be  honouring enablePeriodReplacement incase we're base 64 encoding the tag-values ?

**Comment by katholen** [comment]
@ayansen -> Please review my changes

File: src/main/scala/com/expedia/www/haystack/commons/entities/encodings/Encoding.scala [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.expedia.www.haystack.commons.entities.encodings
+
+trait Encoding {
```
Comment (ayansen) [inline src/main/scala/com/expedia/www/haystack/commons/entities/encodings/Encoding.scala]: I'd suggest calling it Encoder

File: src/main/scala/com/expedia/www/haystack/commons/entities/encodings/EncodingFactory.scala [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.expedia.www.haystack.commons.entities.encodings
+
+object EncodingFactory {
```
Comment (ayansen) [inline src/main/scala/com/expedia/www/haystack/commons/entities/encodings/EncodingFactory.scala]: EncoderFactory perhaps

**Pull request merged**